mutiny-core/src/lspclient.rs

```diff
@@ -114,38 +120,64 @@ impl LspClient {
         })
     }
 
-    pub(crate) async fn get_lsp_invoice(&self, bolt11: String) -> anyhow::Result<String> {
+    pub(crate) async fn get_lsp_invoice(&self, bolt11: String) -> Result<String, MutinyError> {
         let payload = ProposalRequest {
             bolt11,
             host: None,
             port: None,
         };
 
-        let proposal_response: ProposalResponse = self
+        let response: reqwest::Response = self
             .http_client
             .post(format!("{}{}", &self.url, PROPOSAL_PATH))
             .json(&payload)
             .send()
             .await
-            .map_err(|_| MutinyError::LspFailure)?
-            .json()
-            .await
-            .map_err(|_| MutinyError::LspFailure)?;
-
-        Ok(proposal_response.jit_bolt11)
+            .map_err(|_| MutinyError::LspGenericError)?;
+
+        // if the status is 500, copy the response body to a string and try to parse as ErrorResponse
```

Comment:
I think it'd be better checking for 200 and parsing the error on any non-successful request

Reply:
Changed